File: Data Structures & Algorithms/boats-to-save-people/submission-0.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        count = 0
        people.sort()
        left = 0
        right = len(people) - 1
        while(left <= right):
            sum = people[left] + people[right]   
            if sum <= limit:
                left += 1
                right -= 1
            elif sum > limit:
                right -= 1
            count += 1
        return count

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.swift
class Solution {
    func findMin(_ nums: [Int]) -> Int {
        var left = 0
        var right = nums.count - 1
        while(left < right) {
            let mid = left + (right - left) / 2
            if nums[right] < nums[mid] {
                left = mid + 1
            } else {
                right = mid
            }
        }
        return nums[left]
    }
}
